File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
import java.util.*;

class Solution {
    public long solution(long n) {
        String[] list = String.valueOf(n).split("");
        String s = "";
        
        Arrays.sort(list);
        
        for(int i = list.length - 1; i > -1; i--){
            s = s + list[i];
        }
        
        
        return Long.parseLong(s);
    }
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.go
import (
    "sort"
    "strconv"
    "strings"
)

func solution(n int64) int64 {
    s := strconv.FormatInt(n, 10)

    digits := strings.Split(s, "")

    sort.Slice(digits, func(i, j int) bool {
        return digits[i] > digits[j]
    })

    sortedStr := strings.Join(digits, "")
    result, _ := strconv.ParseInt(sortedStr, 10, 64)

    return result
}
